src/ofdm.c

```diff
@@ -661,10 +681,22 @@ static int est_timing(struct OFDM *ofdm, complex float *rx, int length) {
         corr_en = 0.0f;
 
         for (j = 0; j < (ofdm_m + ofdm_ncp); j++) {
-            csam = conjf(ofdm->pilot_samples[j]);
-
-            corr_st = corr_st + (rx[i + j         ] * csam);
-            corr_en = corr_en + (rx[i + j + SFrame] * csam);
+            /* roll frequency shift into inner loop to save memory */
+            switch(fcoarse) {
+            case -40:
+                csam = conjf(ofdm_wval[j % ofdm_nval]*ofdm->pilot_samples[j]);
+                break;
+            case 0:
+                csam = conjf(ofdm->pilot_samples[j]);
+                break;
+            case 40:
+                csam = ofdm_wval[j % ofdm_nval]*conjf(ofdm->pilot_samples[j]);
+                break;
+            default:
+                assert(0);
```

Comment:
@srsampson Hi - did you notice this code, that doesn't need wvec any more?  Would be interested in yr thoughts....

Reply:
I like how it goes frame by frame instead of across the whole RX buffer. This also doesn't need the ofdm_nval contraption, as the 160 samples is less than the 200 samples of wvec[].

Reply:
I was going to flip the wvec[] const array file to be non-conjugate and adjust the functions, but my math kind of blew up, so I left it alone. I was trying to reduce the number of conjf() calls, but I don't think there would be much change.

Reply:
> I like how it goes frame by frame instead of across the whole RX buffer. This also doesn't need the ofdm_nval contraption, as the 160 samples is less than the 200 samples of wvec[].

Oh cool I hadn't noticed that - was more interested in saving stack usage.  However I guess this is more inner-llop operations overall, we are down shifting the same samples again and again.  It's also a slightly different algorithm and the phase of the down shift changes for each ```corr[i]``` samples.  However as we take the abs() value it's probably OK.  And I guess tofdm passes.

I have tested the Octave code quite carefully with several simulations.  I might write some more challenging tests for the C version as well, to make sure it's getting good acquisition performance.